src/components/vm/consoles/vnc.tsx

```diff
@@ -470,9 +489,20 @@ export class VncActive extends React.Component<VncActiveProps, VncActiveState> {
             this.observer.disconnect();
     }
 
-    onDisconnected(clean: boolean) { // server disconnected
+    async onDisconnected(clean: boolean) { // server disconnected
         console.info('Connection lost: ', clean ? "clean" : "unclean");
-        this.props.state.setConnected(false);
+        let reason = this.props.state.failure_reason;
+        if (clean)
+            reason = null;
+        else if (!reason) {
+            // This might be TLS.
+            const qemu_conf = await readQemuConf();
+            if (qemu_conf.vnc_tls)
+                reason = _("VNC with TLS is not supported by the in-page viewer");
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2038-5a05fe26-20250916-083706-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)